src/sentry/pipeline/base.py

```diff
@@ -228,11 +229,27 @@ def bind_state(self, key: str, value: Any) -> None:
 
         self.state.data = data
 
-    def fetch_state(self, key: str | None = None) -> Any | None:
+    def _fetch_state(self, key: str | None = None) -> Any | None:
         data = self.state.data
         if not data:
             return None
         return data if key is None else data.get(key)
 
+    def fetch_state(self, key: str | None = None) -> Any | None:
+        step_index = self.state.step_index
+        if step_index >= len(self.pipeline_views):
+            return self._fetch_state(key)
+        view = self.pipeline_views[step_index]
+        if isinstance(view, NestedPipelineView):
+            # Attempt to surface state from a nested pipeline
+            nested_pipeline = view.pipeline_cls(
+                organization=self.organization,
+                request=self.request,
+                provider_key=view.provider_key,
+                config=view.config,
+            )
+            return nested_pipeline.fetch_state(key)
+        return self._fetch_state(key)
```

Comment:
I thought there was some way to access the nested state from the parent pipeline already no?

Reply:
@evanpurkhiser no way to access it **_from outside the pipeline_** since it's behind the current view in `self.pipeline_views`.